### PR TITLE
BP-984 - stream logs from message-server to backend

### PR DESCRIPTION
**Pull request by erezz-mov-ai**
[BP-984](https://movai.atlassian.net/browse/BP-984) - stream logs from message-server to backend

[BP-984]: https://movai.atlassian.net/browse/BP-984?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: backend/core/log_streamer/client.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from aiohttp import web
+from logging import Logger
+import uuid
+
+from movai_core_shared.logger import Log
+from movai_core_shared.messages.log_data import LogRequest
+
+from backend.core.log_streamer.filter import LogFilter
+
+class Client:
```
Comment (engdorm) [inline backend/core/log_streamer/client.py:21]: need better naming

File: backend/core/log_streamer/client.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    async def send_msg(self, msg: LogRequest):
+        try:
+            if self._filter.filter_msg(msg):
+                await self._sock.send_json(msg)
+        except ValueError as err:
+            self.logger.error(err.__str__())
+        except RuntimeError as err:
+            self.logger.error(err.__str__())
+        except TypeError as err:
```
Comment (engdorm) [inline backend/core/log_streamer/client.py:58]: if they all with the same logic why did you separate to types

File: backend/core/log_streamer/logs_streamer.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+"""
+from movai_core_shared.core.zmq_server import ZMQServer
+from movai_core_shared.envvars import LOG_STREAMER_BIND_ADDR
+from movai_core_shared.logger import Log
+from movai_core_shared.messages.log_data import LogRequest
+
+
+class LogsStreamer(ZMQServer):
+
+    def __init__(self, debug: bool = False) -> None:
```
Comment (engdorm) [inline backend/core/log_streamer/logs_streamer.py:20]: add docstring please
not sure about debug

and maybe you can use the envvar that we have for debug

File: backend/core/log_streamer/logs_streamer.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            #for client in self._clients:
+            #    client.send_msg(request.req_data.log_fields.message)
+            return {}
+        except Exception as error:
+            self._logger.error(str(error))
+            return {}
+
+
+log_streamer = LogsStreamer()
+log_streamer.run()
```
Comment (engdorm) [inline backend/core/log_streamer/logs_streamer.py:39]: why not with main function

File: backend/helpers/rest_helpers.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    """fetches the params from the request and returns them in a dictionary.
+
+    Args:
+        request (dict): The request with the params.
+
+    Returns:
+        dict: A dictionary of params and their value.
+    """
+    params = {}
+    for param in request.query_string.split("&"):
```
Comment (engdorm) [inline backend/helpers/rest_helpers.py:21]: request is marked as dict

**Comment by engdorm** [comment]
go over code smells

**Comment by erezz-mov-ai** [comment]
will merge with 985

**Pull request closed without merging**